### PR TITLE
libs/autofile/group: don't panic when checking Head size

### PR DESCRIPTION
**Pull request by melekes**
* [ ] ~~Updated all relevant documentation in docs~~
* [x] Updated all code comments where relevant
* [x] Wrote tests (partially)
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: consensus/wal.go [hunk 1]
```diff
@@ -95,6 +96,11 @@ func (wal *baseWAL) Group() *auto.Group {
 	return wal.group
 }
 
+func (wal *baseWAL) SetLogger(l log.Logger) {
```
Comment (ebuchman) [inline consensus/wal.go]: Do we need to call this from somewhere?

File: libs/autofile/autofile.go [hunk 2]
```diff
@@ -158,23 +168,22 @@ func (af *AutoFile) openFile() error {
...
+// get stats or open file.
+// Opens AutoFile if needed.
 func (af *AutoFile) Size() (int64, error) {
 	af.mtx.Lock()
 	defer af.mtx.Unlock()
 
 	if af.file == nil {
-		err := af.openFile()
-		if err != nil {
-			if err == os.ErrNotExist {
```
Comment (ebuchman) [inline libs/autofile/autofile.go]: Why not keep this?

Comment (melekes) [reply libs/autofile/autofile.go]: see commit msg here  https://github.com/tendermint/tendermint/pull/2764/commits/dca010d8dfed2fa5f533a01361fb935eab410141

File: consensus/wal.go (hunk 1 above)
Comment (melekes) [reply consensus/wal.go]: We do call it, already. This method overrides a default one.

**Pull request merged**